torch/_inductor/decomposition.py

```diff
@@ -269,6 +269,62 @@ def angle(x):
         return ret + nan
 
 
+@register_decomposition([aten.add])
+def add(x, y, **kwargs):
+    x_is_complex_tensor = torch.is_tensor(x) and x.is_complex()
+    y_is_complex_tensor = torch.is_tensor(y) and y.is_complex()
+    if not x_is_complex_tensor and not y_is_complex_tensor:
+        return NotImplemented
+    r = y.real
+    i = y.imag if y_is_complex_tensor else 0.0
+    alpha = kwargs.get("alpha")
+    if alpha is not None:
+        r *= alpha
+        i *= alpha
```

Comment:
Let's just always decompose this (move it to the top) then recursively call add to decompose the rest.